### ISSUE TITLE
[Bug]: CodePipeline action references the old CodeStarConnection ARN

### ISSUE THREAD

**Issue by ishiguro-junya** [hashicorp/terraform-provider-aws#38611]
### Terraform Core Version

1.8.5

### AWS Provider Version

5.55.0

### Affected Resource(s)

* aws_codepipeline
* aws_codestarconnections_connection

### Expected Behavior

CodeStar connections will be renamed to CodeConnections.
As a result, when you create a new connection, the ARN will change to codeconnections instead of codestar-connections.
However, connections created after 7/1 will need to reference the new ARN.
https://docs.aws.amazon.com/ja_jp/dtconsole/latest/userguide/rename.html

### Actual Behavior

Referencing the old ARN

### Terraform Configuration Files

```hcl
data "aws_iam_policy_document" "codepipeline" {
  statement {
    effect    = "Allow"
    resources = ["*"]

    actions = [
      "s3:PutObject",
      "s3:GetObject",
      "s3:GetObjectVersion",
      "s3:GetBucketVersioning",
      "iam:PassRole",
      "codestar-connections:UseConnection",
    ]
  }
}

resource "aws_s3_bucket" "artifact" {
  bucket        = "artifact"
  force_destroy = true

  lifecycle_rule {
    enabled = true

    expiration {
      days = "180"
    }
  }
}

resource "aws_codestarconnections_connection" "github_connection" {
  name               = "xxxx"
  provider_type = "GitHub"
}

resource "aws_codepipeline" "codepipeline" {
  name     = "codepipeline"
  role_arn = module.codepipeline_role.iam_role_arn

  stage {
    name = "Source"

    action {
       name             = "Source"
       category         = "Source"
       owner            = "AWS"
       provider         = "CodeStarSourceConnection"
       version          = 1
       output_artifacts = ["Source"]

       configuration = {
         ConnectionArn        = aws_codestarconnections_connection.github_connection.arn
         FullRepositoryId     = "xxxx/xxxx"
         BranchName           = "develop"
       }
     }

    artifact_store {
      location = aws_s3_bucket.artifact.id
      type     = "S3"
    }
  }
```

### Steps to Reproduce

Create only the aws_codestarconnections_connection resource and enable the status with an account with GitHub owner permissions.
Then deploy other resources, including codepipeline.

### Would you like to implement a fix?

None

**Comment by ishiguro-junya** [comment]
Pull request in preparation
https://github.com/ishiguro-junya/terraform-provider-aws/tree/f-codeconnections

**Comment by ishiguro-junya** [comment]
This is because CodeStar Connections has been renamed to CodeConnection, and the ARN of CodeConnection cannot be referenced due to the change in the service prefix.
This can be resolved by adding a CodeConnection resource.
This issue will be closed as it will be addressed in #36671.
https://docs.aws.amazon.com/dtconsole/latest/userguide/rename.html
